src/store/modules/conversations.js

```diff
@@ -1,243 +1,119 @@
-      error: null,
-    },
-    receiveStatus: {
-      isWaiting: false,
-      error: null,
-      success: false,
-    },
-    receiveMoreStatus: {
-      isWaiting: false,
-      error: null,
-    },
   }
 }
 
-export const state = initialState()
-
-export const getters = {
-  enrichMessage: (state, getters, rootState, rootGetters) => message => {
-    return {
-      ...message,
-      author: rootGetters['users/get'](message.author),
-    }
-  },
-  activeMessages: (state, getters) => {
-    if (!state.activeConversationId) return []
-    let messages = state.messages[state.activeConversationId] || []
-    return messages.map(getters.enrichMessage)
-  },
-  sendStatus: state => state.sendStatus,
-  receiveStatus: state => state.receiveStatus,
-  receiveMoreStatus: state => state.receiveMoreStatus,
-  canLoadMore: (state, getters, rootState, rootGetters) => {
-    return typeof state.cursors[state.activeConversationId] === 'string'
-  },
-}
-
-export const actions = {
-
-  async setActive ({ commit, dispatch }, conversation) {
-    commit(types.SET_ACTIVE, { conversationId: conversation.id })
-    commit(types.RECEIVE_CONVERSATION, { conversation })
-    await dispatch('fetchMessages', conversation.id)
-  },
-
-  async clearActive ({ commit }) {
-    commit(types.CLEAR_ACTIVE)
-  },
-
-  async sendMessage ({ commit, state, dispatch }, messageData) {
-    commit(types.REQUEST_SEND_MESSAGE)
-    let message
-    try {
-      message = await messageAPI.create({
-        content: messageData,
-        conversation: state.activeConversationId,
-      })
-    }
-    catch (error) {
-      onlyHandleAPIError(error, data => commit(types.RECEIVE_SEND_MESSAGE_ERROR, data))
-      return
-    }
-    commit(types.RECEIVE_SEND_MESSAGE)
-    commit(types.RECEIVE_MESSAGE, { message })
-  },
-
-  async receiveMessage ({ commit, state, getters }, message) {
-    // only add if messages doesn't exist yet
-    if (!getters.activeMessages.find(e => e.id === message.id)) {
-      commit(types.RECEIVE_MESSAGE, { message })
-    }
+export default {
+  namespaced: true,
+  modules: { meta: createMetaModule() },
+  state: initialState(),
+  getters: {
+    enrichMessage: (state, getters, rootState, rootGetters) => message => {
+      return {
+        ...message,
+        author: rootGetters['users/get'](message.author),
+      }
+    },
+    activeMessages: (state, getters) => {
+      return (state.messages[state.activeConversationId] || []).map(getters.enrichMessage)
+    },
+    active: (state, getters) => {
+      const id = state.activeConversationId
+      if (!id) return
+
+      const canLoadMore = typeof state.cursors[state.activeConversationId] === 'string'
+      return {
+        messages: getters.activeMessages,
+        canLoadMore,
+        ...metaStatusesWithId(getters, ['send', 'fetch', 'fetchMore'], id),
+      }
+    },
   },
+  actions: {
+    ...withMeta({
+      async send ({ commit, state, dispatch }, conversationId, messageData) {
```

Comment:
Oh, I didn't know you pass two args? Is it supported?

Reply:
I thought it works, then I tried again, it fails. Will pass an object instead.